File: app/src/main/java/com/github/gfx/helium/util/LoadingAnimation.java
package com.github.gfx.helium.util;

import android.view.View;

import javax.annotation.ParametersAreNonnullByDefault;
import javax.inject.Inject;

@ParametersAreNonnullByDefault
public class LoadingAnimation {

    static long DURATION = 1000;

    @Inject
    public LoadingAnimation() {
    }

    public void start(final View view) {
        view.animate()
                .alpha(0.3f)
                .setDuration(DURATION)
                .withEndAction(new Runnable() {
                    @Override
                    public void run() {
                        view.animate()
                                .alpha(0.9f)
                                .setDuration(DURATION)
                                .withEndAction(new Runnable() {
                                    @Override
                                    public void run() {
                                        start(view);
                                    }
                                })
                                .start();
                    }
                })
                .start();
    }

    public void cancel(final View view) {
        view.clearAnimation();
    }
}

File: app/src/main/java/com/github/gfx/helium/util/LoadingAnimation.kt
package com.github.gfx.helium.util

import android.view.View

import javax.annotation.ParametersAreNonnullByDefault
import javax.inject.Inject

@ParametersAreNonnullByDefault
class LoadingAnimation
@Inject
constructor() {

    fun start(view: View) {
        view.animate().alpha(0.3f).setDuration(DURATION).withEndAction { view.animate().alpha(0.9f).setDuration(DURATION).withEndAction { start(view) }.start() }.start()
    }

    fun cancel(view: View) {
        view.clearAnimation()
    }

    companion object {

        internal var DURATION: Long = 1000
    }
}
